shotover/src/server.rs

```diff
@@ -277,6 +285,51 @@ impl<C: CodecBuilder + 'static> TcpCodecListener<C> {
             backoff *= 2;
         }
     }
+    async fn check_hotreload(
+        rx: &mut UnboundedReceiver<HotReloadListenerRequest>,
+    ) -> Option<HotReloadListenerRequest> {
+        rx.recv().await
+    }
+
+    /// Handle hot reload request by extracting file descriptor and responding
+    async fn handle_hot_reload_request(&mut self, request: HotReloadListenerRequest) {
+        let response = if let Some(ref listener) = self.listener {
+            // Extract the file descriptor from the TcpListener
+            let fd = listener.as_raw_fd();
+
+            let port = match listener.local_addr() {
+                Ok(addr) => addr.port(),
+                Err(_) => {
+                    // Fallback: split once from the right to support hostnames and [IPv6]:port formats.
+                    self.listen_addr
```

Comment:
This is a good idea to use this as a fallback.
But we should take it one step further and just always parse the port from the `listen_addr` field. Relying on a rarely used fallback means you might have bugs hidden in the fallback implementation.

Also we should use the parser in the standard library to do this: https://doc.rust-lang.org/std/net/enum.SocketAddr.html